### PR TITLE
Task 15: Add loading on login and signup

### PR DESCRIPTION
**Pull request by arikamat**

### CODE REVIEW

**Comment by arikamat** [comment]
Need to add tests before approval

- Test that the loading spinner appears during API requests.
- Test error handling by simulating failed API responses.

**Review by jerkur22 — APPROVED** [review]
looks good

**Comment by arikamat** [comment]
[Task 15: Implement a Loading Spinner and Error Handling for API Requests](https://trello.com/c/YRsioiFi/16-task-15-implement-a-loading-spinner-and-error-handling-for-api-requests)

**Pull request merged**
